From f5b361809400e14bce5938e0df71f751b023e7a1 Mon Sep 17 00:00:00 2001
From: Daniel Krol <orblivion@gmail.com>
Date: Wed, 27 Oct 2021 20:27:46 -0400
Subject: [PATCH 08/13] Pass full userInfo into setAuthorInfo() /
 setAuthorStyle() hook.

This allows the plugin to determine the author name and color in a sane way. Without this, ep_author_neat currently scrapes the DOM to find this information, which doesn't work sometimes.

(msg copied from Kenton)
---
 src/static/js/collab_client.js | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/static/js/collab_client.js b/src/static/js/collab_client.js
index e507d6b0..434c76ce 100644
--- a/src/static/js/collab_client.js
+++ b/src/static/js/collab_client.js
@@ -321,10 +321,10 @@ const getCollabClient = (ace2editor, serverVars, initialUserInfo, options, _pad)
   };
 
   const tellAceActiveAuthorInfo = (userInfo) => {
-    tellAceAuthorInfo(userInfo.userId, userInfo.colorId);
+    tellAceAuthorInfo(userInfo.userId, userInfo.colorId, userInfo);
   };
 
-  const tellAceAuthorInfo = (userId, colorId, inactive) => {
+  const tellAceAuthorInfo = (userId, colorId, userInfo, inactive) => {
     if (typeof colorId === 'number') {
       colorId = clientVars.colorPalette[colorId];
     }
@@ -334,16 +334,18 @@ const getCollabClient = (ace2editor, serverVars, initialUserInfo, options, _pad)
       editor.setAuthorInfo(userId, {
         bgcolor: cssColor,
         fade: 0.5,
+        userInfo,
       });
     } else {
       editor.setAuthorInfo(userId, {
         bgcolor: cssColor,
+        userInfo,
       });
     }
   };
 
   const fadeAceAuthorInfo = (userInfo) => {
-    tellAceAuthorInfo(userInfo.userId, userInfo.colorId, true);
+    tellAceAuthorInfo(userInfo.userId, userInfo.colorId, userInfo, true);
   };
 
   const getConnectedUsers = () => valuesArray(userSet);
@@ -351,7 +353,7 @@ const getCollabClient = (ace2editor, serverVars, initialUserInfo, options, _pad)
   const tellAceAboutHistoricalAuthors = (hadata) => {
     for (const [author, data] of Object.entries(hadata)) {
       if (!userSet[author]) {
-        tellAceAuthorInfo(author, data.colorId, true);
+        tellAceAuthorInfo(author, data.colorId, data, true);
       }
     }
   };
-- 
2.20.1

